src/Umbraco.AuthorizedServices/Services/Implement/AuthorizationRequestSender.cs

```diff
@@ -16,7 +16,9 @@ public async Task<HttpResponseMessage> SendRequest(ServiceDetail serviceDetail,
 
         var url = serviceDetail.GetTokenHost() + serviceDetail.RequestTokenPath;
 
-        if (serviceDetail.AuthorizationRequestRequiresAuthorizationHeaderWithBasicToken)
+        if ((serviceDetail.AuthenticationMethod == AuthenticationMethod.OAuth2ClientCredentials
+                && serviceDetail.ClientCredentialsProvision == ClientCredentialsProvision.AuthHeader)
+            || serviceDetail.AuthorizationRequestRequiresAuthorizationHeaderWithBasicToken)
```

Comment:
Should add new test in `AuthorizationRequestSenderTests` called `SendRequest_ForOAuth2ClientCredentialsWithClientCredentialsProvisionInHeader_SendsRequest`

Reply:
Updated